scripts/adopt-plane.test.sh

```
@@ -0,0 +1,86 @@
+
+# The suite owns a private temp root: every child instance scopes its
+# transient files under it, so concurrent suites (the self-test and probe
+# lanes run adopt-plane suites in parallel) cannot interfere with each other.
+SUITE_TMP=$(mktemp -d) || exit 1
+export TMPDIR=$SUITE_TMP
+trap 'rm -rf "$SUITE_TMP"' EXIT
+
+# Number of entries in the suite's own temp root: child instances must leave
+# it exactly as the suite arranges it.
+suite_entries() {
+  ls -A "$SUITE_TMP" | grep -c .
```

Shellcheck (0.11.0)

[warning] Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.

(SC2010)